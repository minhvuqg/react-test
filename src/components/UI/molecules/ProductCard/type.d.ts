export interface IProps {
  product: Product.Infor
  onClickAddToCart?: (product: Product.Infor) => void
}
