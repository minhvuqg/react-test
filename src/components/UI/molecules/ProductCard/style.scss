.product-card {
  max-width: 500px;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 8px;
  margin-right: 16px;

  &:last-of-type {
    margin-right: 0;
  }

  .image {
    width: 300px;
    height: 200px;

    img {
      width: 100%;
      height: 100%;
      object-fit: contain;
    }
  }
}
