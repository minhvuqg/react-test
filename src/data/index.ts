export const productList: Product.Infor[] = [
  {
    id: '1',
    image: '/imgs/products/iphone.png',
    name: 'Iphone',
    price: 1000,
    description:
      'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Mollitia doloremque architecto quia aliquam? Assumenda a, ipsam quidem enim neque saepe corrupti, fugiat exercitationem vero, optio dolorem alias maxime dolore omnis.'
  },
  {
    id: '2',
    image: '/imgs/products/macbook.jpg',
    name: 'Macbook pro 14 inch 2021 32gb',
    price: 2000,
    description:
      'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Mollitia doloremque architecto quia aliquam? Assumenda a, ipsam quidem enim neque saepe corrupti, fugiat exercitationem vero, optio dolorem alias maxime dolore omnis.'
  },
  {
    id: '3',
    image: '/imgs/products/watch.jpg',
    name: 'Watch',
    price: 500,
    description:
      'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Mollitia doloremque architecto quia aliquam? Assumenda a, ipsam quidem enim neque saepe corrupti, fugiat exercitationem vero, optio dolorem alias maxime dolore omnis.'
  }
]
