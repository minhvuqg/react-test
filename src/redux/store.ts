import { combineReducers } from 'redux'
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'

import productSlice from './reducers/productSlice'

const customizedMiddleware: any = getDefaultMiddleware({
  serializableCheck: false
})

const persistedReducer: any = combineReducers({
  cart: productSlice
})

export default configureStore({
  reducer: persistedReducer,
  middleware: customizedMiddleware
})
