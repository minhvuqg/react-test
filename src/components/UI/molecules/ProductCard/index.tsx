import { FC } from 'react'
import { useHistory } from 'react-router-dom'

import { convertToCurrency } from 'helpers/string'

import { Button } from 'components/UI/atoms/Button'

import { IProps } from './type'

import './style.scss'

export const ProductCard: FC<IProps> = ({
  product,
  onClickAddToCart = (): void => { }
}: IProps): JSX.Element => {
  const history: any = useHistory()

  const goToDetailPage: any = (): void => {
    history.push(`/products/${product.id}`)
  }

  const handleOnClickAddToCart: any = (): void => {
    onClickAddToCart(product)
  }

  return (
    <div className="product-card">
      <div className="image">
        <img data-testid="img" src={product.image} alt={product.name} />
      </div>

      <span data-testid="name" className="m-t-32">
        {product.name}
      </span>

      <span data-testid="price" className="m-t-8 -text-bold">
        {convertToCurrency(product.price)}
      </span>

      <div className="actions m-t-32">
        <Button
          data-testid="moreInfoBtn"
          className="m-r-16"
          onClick={goToDetailPage}
        >
          More info
        </Button>

        <Button
          data-testid="addToCartBtn"
          primary={true}
          onClick={handleOnClickAddToCart}
        >
          Add to cart
        </Button>
      </div>
    </div>
  )
}
