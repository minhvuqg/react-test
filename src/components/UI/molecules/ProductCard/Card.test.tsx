import TestRenderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { ProductCard } from '../ProductCard'
import store from 'redux/store'
import { productList } from 'data'

it('render card correctly', () => {
  const cardEl = TestRenderer.create(
    < Provider store={store} >
      <ProductCard product={productList[0]} />
    </Provider >).toJSON()
  expect(cardEl).toMatchSnapshot()
})
