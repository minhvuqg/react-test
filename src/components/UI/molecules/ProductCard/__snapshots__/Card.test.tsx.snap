// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render card correctly 1`] = `
<div
  className="product-card"
>
  <div
    className="image"
  >
    <img
      alt="Iphone"
      data-testid="img"
      src="/imgs/products/iphone.png"
    />
  </div>
  <span
    className="m-t-32"
    data-testid="name"
  >
    Iphone
  </span>
  <span
    className="m-t-8 -text-bold"
    data-testid="price"
  >
    $1,000.00
  </span>
  <div
    className="actions m-t-32"
  >
    <button
      className="-custom-btn m-r-16 "
      data-testid="moreInfoBtn"
      onClick={[Function]}
    >
      More info
    </button>
    <button
      className="-custom-btn undefined -primary"
      data-testid="addToCartBtn"
      onClick={[Function]}
    >
      Add to cart
    </button>
  </div>
</div>
`;
